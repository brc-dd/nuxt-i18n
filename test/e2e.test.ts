import { fileURLToPath } from 'node:url'
import { describe, expect, it } from 'vitest'
import { $fetch, setup } from '@nuxt/test-utils'

describe('nuxt-i18n', async () => {
  await setup({
    server: true,
    rootDir: fileURLToPath(new URL('../playground', import.meta.url)),
  })

  it('renders the test page correctly for default locale', async () => {
    const html = await $fetch('/en/test')
    // Extract content from <script type="text/test-result">...</script> but make sure no other script are inside
    const content = html.match(/<script\s+type="text\/test-result">(.*?)<\/script>/s)?.[1]
    expect(content).toMatchSnapshot()
  })

  it('renders the test page correctly for "de" locale', async () => {
    const html = await $fetch('/de/test')
    const content = html.match(/<script\s+type="text\/test-result">(.*?)<\/script>/s)?.[1]
    expect(content).toMatchSnapshot()
  })
})
