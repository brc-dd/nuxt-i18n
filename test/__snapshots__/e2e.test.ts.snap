// Vitest Snapshot v1

exports[`nuxt-i18n > builds routes tree 1`] = `
[
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "test-composables___en",
    "path": "/en/test/composables",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "test-composables___de",
    "path": "/de/test/composables",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "test-i18n___en",
    "path": "/en/test/i18n",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "test-i18n___de",
    "path": "/de/test/i18n",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "test-routes___en",
    "path": "/en/test/routes",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "test-routes___de",
    "path": "/de/test/routes",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "about___en",
    "path": "/en/about",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "about___de",
    "path": "/de/about",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "test___en",
    "path": "/en/test",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "test___de",
    "path": "/de/test",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "index___en",
    "path": "/",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "index___de",
    "path": "/de",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "id___de",
    "path": "/de/:id(.*)*",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
  {
    "children": [],
    "enterCallbacks": {},
    "instances": {},
    "leaveGuards": {},
    "meta": {},
    "name": "id___en",
    "path": "/:id(.*)*",
    "props": {
      "default": false,
    },
    "updateGuards": {},
  },
]
`;

exports[`nuxt-i18n > contains i18n data 1`] = `
{
  "defaultLocale": "en",
  "locale": "en",
  "locales": [
    "en",
    "de",
  ],
  "message": "Home",
}
`;

exports[`nuxt-i18n > renders message for "de" 1`] = `"Start"`;

exports[`nuxt-i18n > renders message for default locale 1`] = `"Home"`;

exports[`nuxt-i18n > returns composables data for "de" 1`] = `
{
  "useLocalizedPath": [
    "/en/test/composables",
    "/de/test/composables",
  ],
  "useRouteLocale": "de",
}
`;

exports[`nuxt-i18n > returns composables data for default locale 1`] = `
{
  "useLocalizedPath": [
    "/en/test/composables",
    "/de/test/composables",
  ],
  "useRouteLocale": "en",
}
`;
